### ISSUE TITLE
Does not handle multi-threaded token invalidation

### ISSUE THREAD

**Issue by ruswerner** [sedenardi/zoomapi#14]
Just in case someone else was using this library with JWT and has moved to Server-to-server Oauth (v2.x)... this library does not support multi-threaded server applications with Oauth _out of the box_. By default, Zoom will invalidate any previous oauth tokens when requesting a new one. This library always requests a new token. Thus if you have a multi-threaded or multi-instance server application, you will get random race conditions with "Invalid access token" errors in your logs. See this zoom discussion for more context:

https://devforum.zoom.us/t/server-to-server-oauth-annoying-token-invalidation/72070/15

tl;dr
You need to implement your own thread-safe shared token service, and then you can use the `onGetAccessToken` feature of this library to avoid fetching the token directly from zoom and instead fetch it from your token service.

**Comment by sedenardi** [comment]
I intentionally left it up to the consumers of this library how they want to implement locking when fetching a token for the same Zoom account. If there are multiple instances of an application then it wouldn't be possible to achieve within this library alone.
